addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -2583,6 +2584,131 @@ export class OdooEditor extends EventTarget {
     // TABLE MANAGEMENT
     // ================
 
+    /**
+     * Sets selection in table cells in order to select the cells
+     * rectangularly when pressing shift + arrow key (just like
+     * google docs).
+     * 
+     * @private
+     * @param {KeyboardEvent}
+     */
+    _setSelectionInTable(ev) {
```

Comment:
Hello @adsa-odoo i would like to add one more thing.
This method is doing a lot of things. It would be easier to understand and maintain if it was broken down into smaller, more manageable functions. Each function should do one thing and do it well.Like your code handle single cell selection and scenario for multiple cell selection here only try to break them to separate functions.